src/main/java/com/amazonaws/encryptionsdk/keyrings/AwsKmsSymmetricMultiCmkKeyringBuilder.java

```diff
@@ -0,0 +1,184 @@
+        return this;
+    }
+
+    /**
+     * A list of {@link AwsKmsCmkId}s in ARN, CMK Alias, or ARN Alias format identifying AWS KMS CMKs
+     * used for encrypting and decrypting data keys.
+     *
+     * @param childKeyNames The list of key names identifying AWS KMS CMKs.
+     * @return The AwsKmsSymmetricMultiCmkKeyringBuilder, for method chaining
+     */
+    public AwsKmsSymmetricMultiCmkKeyringBuilder keyNames(List<AwsKmsCmkId> childKeyNames) {
+        this.childKeyNames = childKeyNames;
+        return this;
+    }
+
+    /**
+     * An {@link AwsKmsCmkId} in ARN, CMK Alias, or ARN Alias format that identifies a
+     * AWS KMS CMK responsible for generating a data key, as well as encrypting and
+     * decrypting data keys.
+     *
+     * @param generatorKeyName An {@link AwsKmsCmkId} in ARN, CMK Alias, or ARN Alias format that identifies a
+     *                         AWS KMS CMK responsible for generating a data key, as well as encrypting and
+     *                         decrypting data keys.
+     * @return The AwsKmsSymmetricMultiCmkKeyringBuilder, for method chaining
+     */
+    public AwsKmsSymmetricMultiCmkKeyringBuilder generator(AwsKmsCmkId generatorKeyName) {
+        this.generatorKeyName = generatorKeyName;
+        return this;
+    }
+
+    /**
+     * Constructs the {@code MultiKeyring} of {@code AwsKmsSymmetricKeyring}s.
+     *
+     * @return The {@link MultiKeyring} instance
+     */
+    public MultiKeyring build() {
+        // A mapping of AWS region to DataKeyEncryptionDao
+        final Map<String, DataKeyEncryptionDao> clientMapping = new HashMap<>();
+
+        // First construct the generator keyring
+        Keyring generatorKeyring = null;
+        if (this.generatorKeyName != null) {
+            // If we have an ARN, obtain the region from the ARN
+            // to specify the region of the AWS SDK KMS service client
+            final Optional<Arn> generatorArn = AwsKmsCmkId.getArnFromKeyName(this.generatorKeyName.toString());
+            final String generatorRegion = generatorArn.isPresent() ? generatorArn.get().getRegion() : null;
+            final DataKeyEncryptionDao generatorDao = constructDataKeyEncryptionDao(generatorRegion);
+
+            // Add the client to the mapping with the region key
+            // This prevents re-creating multiple clients for the same region during a single build call
+            final String generatorRegionKey = StringUtils.isBlank(generatorRegion) ? NULL_REGION : generatorRegion;
+            clientMapping.put(generatorRegionKey, generatorDao);
+
+            // Construct the generator keyring
+            generatorKeyring = new AwsKmsSymmetricKeyring(generatorDao, this.generatorKeyName);
+        }
+
+        // Next, construct the child (non-generator) keyrings
+        List<Keyring> childKeyrings = new ArrayList<>();
+        if (this.childKeyNames != null) {
+            for (final AwsKmsCmkId keyName : this.childKeyNames) {
```

Comment:
I don't believe so. Even pre-spec 1.0, I believe the generator *could* have a duplicate. We want to avoid handling de-duplication whenever possible. For example, if we do an exact string match, we might still have duplicates because the generator could be an alias/ key id/qualified ARN, etc.